### PR TITLE
Minor ConfigMap controller improvements

### PR DESCRIPTION
**Pull request by sebsoto**
* Fixes panic
* Refactors EnsureServicesConfigMapExists()

### CODE REVIEW

File: controllers/configmap_controller.go [hunk 1]
```diff
@@ -454,18 +454,20 @@ func (r *ConfigMapReconciler) EnsureServicesConfigMapExists() error {
...
 	data, err := servicescm.Parse(windowsServices.Data)
-	if err != nil || data.ValidateExpectedContent(r.servicesManifest) != nil {
-		if err = r.k8sclientset.CoreV1().ConfigMaps(r.watchNamespace).Delete(context.TODO(), windowsServices.Name,
-			meta.DeleteOptions{}); err != nil {
-			return err
-		}
-		r.log.Info("Deleted invalid resource", "ConfigMap",
-			kubeTypes.NamespacedName{Namespace: r.watchNamespace, Name: servicescm.Name})
-		return r.createServicesConfigMapOnBootup()
+	if err == nil && data.ValidateExpectedContent(r.servicesManifest) != nil {
```
Comment (saifshaikh48) [inline controllers/configmap_controller.go:458]: Shouldn't the second clause expect no error:  `data.ValidateExpectedContent(r.servicesManifest) == nil`

**Review by aravindhp — APPROVED** [review]
Thanks for fixing this, @sebsoto

**Pull request merged**